### ISSUE TITLE
Custom loss function

### ISSUE THREAD

**Issue by lfbsdev** [jamesdolezal/slideflow#200]
## Feature

A custom loss function at ModelParams.
## Pitch

That would enable further possibilities of replacing common digital pathology ML pipelines with slideflow.
## Alternatives

There isn't any alternative to this I know apart from training without slideflow.

**Comment by jamesdolezal** [comment]
Thanks for the idea and your pull request - I've merged with the branch `custom_loss`, we can continue work from there.

Custom loss support should now be working for Tensorflow/Keras with 561024f. [Here's a script](https://gist.github.com/jamesdolezal/994e60d7a570c70fabb22a0eef5c388e) that uses the TestSuite for quick testing. Use the script by passing a directory to some test slides. It will extract tiles & test the custom categorical and linear losses:

```bash
python custom_loss_test.py --slides=/path/to/slides
```

Linear losses do require that y_true is cast to float.

I'll work on testing it for PyTorch next.

**Comment by jamesdolezal** [comment]
Looks like it works in the PyTorch backend, as well ([gist](https://gist.github.com/jamesdolezal/cce6a25731abd88894c93789b8bf17a7)). 

Let me know if this works for the loss function you were trying to use - if so, I'll go ahead and merge with master.

**Comment by lfbsdev** [comment]
Nice, thank you. I will.

**Comment by jamesdolezal** [comment]
Great! Thanks again.
